src/FlowMeter.cpp

```diff
@@ -58,16 +63,29 @@ void FlowMeter::count() {
     this->_currentPulses++;                                                 //!< this should be called from an interrupt service routine
 }
 
+void FlowMeter::pause() {
+    detachInterrupt(_pin);
+    this->_currentPulses = 0;
+}
+
+void FlowMeter::resume() {
+    attachInterrupt(_pin, _interruptCallback, RISING);
+}
+
 void FlowMeter::reset() {
-    cli();                                                                  //!< going to change interrupt variable(s)
+    detachInterrupt(_pin);                                                  //!< going to change interrupt variable(s)
     this->_currentPulses = 0;                                               //!< reset pulse counter
-    sei();                                                                  //!< done changing interrupt variable(s)
+    attachInterrupt(_pin, _interruptCallback, RISING);                        //!< done changing interrupt variable(s)
 
     this->_currentFrequency = 0.0f;
     this->_currentDuration = 0.0f;
     this->_currentFlowrate = 0.0f;
     this->_currentVolume = 0.0f;
     this->_currentCorrection = 0.0f;
+
+    this->_totalDuration = 0.0f;
```

Comment:
At this point I fail to see the benefit over simply removing the feature to account for a (grand) total altogether, which would then reduce the overall library functionality to that of a simple counter. 

Additionally, this will be a breaking change for other users. 
Also, I don't yet see the difference between totally resetting the meter instance and just instantiating a new meter.

Could you perhaps elaborate why you need a total reset on top of what the upstream `reset()` method already provides?
Particularly what is it that makes you need to reset everything every time, rendering the difference between current and total values pointless? 
Couldn't you simply ignore any totals, not exposing them to the rest of the application and the user?

Reply:
Oh interesting, so you'd recommend instantiating a new meter instead. That makes sense. I think this was an application specific change and I'm trying to remember why I did it. It was over a year ago now.

Reply:
I don't necessarily recommend it.
I'm just wondering what the difference would be. 

My main point is that the difference between current and total values vanishes when always resetting everything, doesn't it?